frontend/packages/dev-console/src/components/pipelines/pipeline-form/SecretsList.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import {
+  ResourceLink,
+  Firehose,
+  FirehoseResult,
+  FirehoseResource,
+} from '@console/internal/components/utils';
+import { SecretModel, ServiceAccountModel } from '@console/internal/models';
+import { SecretType } from '@console/internal/components/secrets/create-secret';
+import { SecondaryStatus } from '@console/shared';
+import { SecretKind, ServiceAccountKind } from '@console/internal/module/k8s';
+import { PIPELINE_SERVICE_ACCOUNT } from '../const';
+import './SecretsList.scss';
+
+type SecretsProps = {
+  secrets?: FirehoseResult<SecretKind[]>;
+  serviceaccounts?: FirehoseResult<ServiceAccountKind>;
```

Comment:
Since, firehose is injecting these values, we will get type error on the component. Hence all the firehoseResult components are made as optional accross the codebase.